File: frontend/actions/company.ts
import {CompanySocialLink} from "@/models";

export async function getCompanySocialLinks() : Promise<CompanySocialLink [] > {


    return  [];
}



/*
const links: ({
    companySocialLink: ({
        socialLink: {
            id: string
            createdAt: Date
            updatedAt: Date
            name: string
            iconLib: string
            iconName: string
        }
    } & {
        id: string
        createdAt: Date
        updatedAt: Date
        url: string
        companyId: string
        socialLinkId: string
    })[]
} & {
    id: string
    createdAt: Date
    updatedAt: Date
    email: string
    name: string
    address: string
    postalCode: string
    city: string
    country: string
    phone: string | null
    siret: string | null
    vatNumber: string | null
    logoUrl: string | null
    }
) | null
*/

File: backend/content-service/src/main/java/com/codeandskills/content_service/domain/model/Company.java
package com.codeandskills.content_service.domain.model;


import jakarta.persistence.*;
import lombok.*;


import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "companies")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Company extends BaseEntity{

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String address;

    @Column(nullable = false)
    private String postalCode;

    @Column(nullable = false)
    private String city;

    @Column(nullable = false)
    private String country;

    @Column(nullable = false)
    private String email;

    private String phone;

    private String siret;

    private String vatNumber;

    private String logoUrl;

    @OneToMany(
            mappedBy = "company",
            cascade = CascadeType.ALL,
            orphanRemoval = true
    )
    @Builder.Default
    private List<CompanySocialLink> companySocialLinks = new ArrayList<>();

}